test/commands/extended/promote_transaction_test.py

```diff
@@ -0,0 +1,369 @@
+      },
+
+      {
+        'depth': [f.Type.CODE_WRONG_TYPE],
+      },
+    )
+
+  def test_fail_depth_too_small(self):
+    """
+    ``depth`` is < 1.
+    """
+    self.assertFilterErrors(
+      {
+        'depth': 0,
+
+        'minWeightMagnitude': 18,
+        'transaction':        TransactionHash(self.trytes1),
+      },
+
+      {
+        'depth': [f.Min.CODE_TOO_SMALL],
+      },
+    )
+
+  def test_fail_min_weight_magnitude_null(self):
+    """
+    ``minWeightMagnitude`` is null.
+    """
+    self.assertFilterErrors(
+      {
+        'minWeightMagnitude': None,
+
+        'depth':        100,
+        'transaction':  TransactionHash(self.trytes1),
+      },
+
+      {
+        'minWeightMagnitude': [f.Required.CODE_EMPTY],
+      },
+    )
+
+  def test_fail_min_weight_magnitude_string(self):
+    """
+    ``minWeightMagnitude`` is a string.
+    """
+    self.assertFilterErrors(
+      {
+        # It's gotta be an int!
+        'minWeightMagnitude': '18',
+
+        'depth':        100,
+        'transaction':  TransactionHash(self.trytes1),
+      },
+
+      {
+        'minWeightMagnitude': [f.Type.CODE_WRONG_TYPE],
+      },
+    )
+
+  def test_fail_min_weight_magnitude_float(self):
+    """
+    ``minWeightMagnitude`` is a float.
+    """
+    self.assertFilterErrors(
+      {
+        # Even with an empty fpart, float values are not valid.
+        'minWeightMagnitude': 18.0,
+
+        'depth':        100,
+        'transaction':  TransactionHash(self.trytes1),
+      },
+
+      {
+        'minWeightMagnitude': [f.Type.CODE_WRONG_TYPE],
+      },
+    )
+
+  def test_fail_min_weight_magnitude_too_small(self):
+    """
+    ``minWeightMagnitude`` is < 1.
+    """
+    self.assertFilterErrors(
+      {
+        'minWeightMagnitude': 0,
+
+        'depth':        100,
+        'transaction':  TransactionHash(self.trytes1),
+      },
+
+      {
+        'minWeightMagnitude': [f.Min.CODE_TOO_SMALL],
+      },
+    )
+
+
+class PromoteTransactionCommandTestCase(TestCase):
+  def setUp(self):
+    super(PromoteTransactionCommandTestCase, self).setUp()
+
+    self.adapter = MockAdapter()
+    self.command = PromoteTransactionCommand(self.adapter)
+
+    self.hash1 = TransactionHash(
+      b'TESTVALUE9DONTUSEINPRODUCTION99999TBPDM9'
+      b'ADFAWCKCSFUALFGETFIFG9UHIEFE9AYESEHDUBDDF'
+    )
+
+    self.hash2 = TransactionHash(
+      b'TESTVALUE9DONTUSEINPRODUCTION99999CIGCCF'
+      b'KIUFZF9EP9YEYGQAIEXDTEAAUGAEWBBASHYCWBHDX'
+    )
+
+  def test_wireup(self):
+    """
+    Verifies that the command is wired-up correctly.
+    """
+    self.assertIsInstance(
+      Iota(self.adapter).promoteTransaction,
+      PromoteTransactionCommand,
+    )
+
+  def test_happy_path(self):
+    """
+    Successfully promoting a bundle.
+    """
+
+    self.adapter.seed_response('checkConsistency', {
+      'state': True,
+    })
+
+    result_bundle = Bundle.from_tryte_strings(self.hash2)
```

Comment:
❔ Is this correct?  I think `Bundle.from_tryte_strings` expects an array of `TransactionTrytes` objects.